### PR TITLE
Remove use of testify mock and testify vendored lib

### PR DESCRIPTION
**Pull request by vdemeester**
Remove the use of testify for mocking stuff (and does a _by-hand_ mock that is not that pretty to look at :sweat_smile:).

Replace #16737 (@jfrazelle) and follow #16733 (very good work from @MHBauer).

Relates (and maybe closes (?)) #16633.

/cc @icecrime @LK4D4 @calavera 

Signed-off-by: Vincent Demeester vincent@sbr.pm

### CODE REVIEW

**Comment by runcom** [comment]
I personally like FakeStore mock you did here

File: pkg/discovery/kv/kv_test.go [hunk 1]
```diff
@@ -122,6 +111,65 @@ func (ds *DiscoverySuite) TestWatch(c *check.C) {
...
+func (s *FakeStore) Exists(key string) (bool, error) {
+	return true, nil
+}
+
+func (s *FakeStore) Watch(key string, stopCh <-chan struct{}) (<-chan *store.KVPair, error) {
+	return nil, nil
+}
+
+func (s *FakeStore) WatchTree(directory string, stopCh <-chan struct{}) (<-chan []*store.KVPair, error) {
+	fmt.Printf("%s\n", directory)
```
Comment (vdemeester) [inline pkg/discovery/kv/kv_test.go]: Oh I should remove that line though... And add comments on the other to say that it's for implementing `Store` and as it's not tested, it does nothing :cat:.

**Comment by MHBauer** [comment]
@vdemeester I would not give me too much credit. I did the lazy thing and used gofmt to do most of the transformations. 

I also like the FakeStore. I definitely learned something here. Thanks.

**Comment by LK4D4** [comment]
LGTM

File: hack/vendor.sh [hunk 2]
```diff
@@ -17,7 +17,6 @@ clone git github.com/mattn/go-sqlite3 v1.1.0
 clone git github.com/microsoft/hcsshim 7f646aa6b26bcf90caee91e93cde4a80d0d8a83e
 clone git github.com/mistifyio/go-zfs v2.1.1
 clone git github.com/stretchr/objx cbeaeb16a013161a98496fad62933b1d21786672
```
Comment (jfrazelle) [inline hack/vendor.sh]: this should go too

Comment (vdemeester) [inline hack/vendor.sh]: ok :wink:

Comment (vdemeester) [inline hack/vendor.sh]: Updated :cat:

**Comment by runcom** [comment]
LGTM

**Comment by aluzzardi** [comment]
What's the point of getting rid of testify?

This PR basically makes the tests entirely useless as it's not testing anything anymore.

**Comment by vdemeester** [comment]
@aluzzardi the point is explained there #16633 "`pkg/discovery` uses `stretchr/testify` for testing, which looks weird, because we don't use any framework for unit-tests and use `go-check` for integration tests.".

That said, I don't get why you are saying that "it's not testing anything anymore".. It just replace the `mock` from testify by a simple/naive mock (there was no need for more complicated with the given code). It still tests the same things that was tested before. The only missing piece would be the last `AssertExpectation` but the behaviour of that could be done too.

Seeing https://github.com/docker/docker/pull/16644#issuecomment-145684214, I get why it can be bothersome to have removed `testify` though, because libkv mock store is not usable anymore and extra work is needed to mock.. 

@LK4D4 @calavera @jfrazelle any opinion on that ?

**Pull request merged**